app/api/entities/entities.js

```diff
@@ -427,12 +448,16 @@ export default {
   },
 
   async getWithoutDocuments(query, select, options = {}) {
-    return model.getUnrestricted(query, select, options);
+    const entities = await model.getUnrestricted(query, select, options);
+    await performNewRelationshipQueries(entities);
```

Comment:
Since this function is modifying the entities passed (mutating them or not), we think its clearer if the function returns the entities, (ideally without mutating the input).

Reply:
Inline with the previous comment: I think we should try to return the object only if the function is pure and doesn't mutate the parameters. Returning a the just-mutated parameter might be misleading by making you think that the function is pure. Will take a look as well.

Reply:
We'll also keep this one as is for now.

Reply:
@daneryl actually I forgot we had decided during a meeting that a good step to fix the original concerns about this particular function was to find a better name. I have committed those changes.

Reply:
@fnocetti i think during the review we commented this but forgot to comment here :(, i think the new name is much better and makes things clearer :+1: